x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java

```diff
@@ -21,29 +22,27 @@
 import java.net.InetSocketAddress;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.Collections;
-import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
 
 public class AuditTrailService {
 
     private static final Logger logger = LogManager.getLogger(AuditTrailService.class);
 
     private static final AuditTrail NOOP_AUDIT_TRAIL = new NoopAuditTrail();
-    private final CompositeAuditTrail compositeAuditTrail;
+    private final @Nullable AuditTrail auditTrail;
     private final XPackLicenseState licenseState;
     private final Duration minLogPeriod = Duration.ofMinutes(30);
     protected AtomicReference<Instant> nextLogInstantAtomic = new AtomicReference<>(Instant.EPOCH);
 
-    public AuditTrailService(List<AuditTrail> auditTrails, XPackLicenseState licenseState) {
-        this.compositeAuditTrail = new CompositeAuditTrail(Collections.unmodifiableList(auditTrails));
+    public AuditTrailService(@Nullable AuditTrail auditTrail, XPackLicenseState licenseState) {
+        this.auditTrail = auditTrail;
```

Comment:
Should we use `NOOP_AUDIT_TRAIL` when the pass-in auditTrail is `null`? I think it's better if we don't have to deal with null within the class.

Reply:
It's definitely better to not have to deal with `null`s. Problem is that `null` here means that auditing is not enabled, while `NOOP` is intended to be used when license level does not allow auditing.
I prefer to not have to deal with the two meanings using the same logger just now...
But thanks for the good point!